### PR TITLE
Add socket stats API documentation

### PR DESCRIPTION
**Pull request by deepikabhavnani**
Mbed OS 5.11 feature.

Dependent on https://github.com/ARMmbed/mbed-os/pull/8592

Todo:

- [x]  Review
- [x]  Doxygen page update once PR is merged
- [x] Verify example link when merged  https://github.com/ARMmbed/mbed-os-examples-docs_only/pull/12

CC @AnotherButler  @kegilbert @SenRamakri

### CODE REVIEW

**Comment by AnotherButler** [comment]
Please see the checklist in the description for a list of what this PR still needs.

File: docs/api/networksocket/SocketStats.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+## SocketStats
+
+Use the SocketStats class to read the network socket statistics. SocketStats Class object is created by "InternetSocket" class when statistics are enabled by `nsapi.socket-stats-enable` config option.
+
+SocketStats class provides static API `mbed_stats_socket_get_each` to get the network socket information. Using this API you can know how many sockets are open, destination IPand data bytes sent and received. Also the timestamp of recent "socket state" change is recorded in `last_change_tick`.
```
Comment (kegilbert) [inline docs/api/networksocket/SocketStats.md]:
```suggestion
The `SocketStats` class provides the static function `mbed_stats_socket_get_each` to get the network socket information for all sockets both currently in use and closed. Using this API you can know how many sockets are open, destination IP addresses, the amount of data sent and received in bytes, the type of socket and the timestamp of recent "socket state" changes recorded in `last_change_tick`.
```

File: docs/api/networksocket/SocketStats.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+Socket Statistics are stored in intenal data buffer and maintained for closed sockets as well. In case internal data buffer is full, socket closed with oldest timestamp will be over-written. Maximum amount of sockets statistics stored can be configured by `nsapi.socket-stats-max-count` config option.
```
Comment (kegilbert) [inline docs/api/networksocket/SocketStats.md]: Link to config page maybe @AnotherButler ?

Comment (deepikabhavnani) [reply docs/api/networksocket/SocketStats.md]: @kegilbert - I don't see network configs here https://github.com/ARMmbed/mbed-os-5-docs/tree/development/docs/reference/configuration

File: docs/api/networksocket/SocketStats.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+## SocketStats
+
+Use the SocketStats class to read the network socket statistics. SocketStats Class object is created by "InternetSocket" class when statistics are enabled by `nsapi.socket-stats-enable` config option.
+
+The `SocketStats` class provides the static function `mbed_stats_socket_get_each` to get the network socket information for all sockets both currently in use and closed. Using this API you can know how many sockets are open, destination IP addresses, the amount of data sent and received in bytes, the type of socket and the timestamp of recent "socket state" changes recorded in `last_change_tick`.
```
Comment (SenRamakri) [inline docs/api/networksocket/SocketStats.md]: Please change "Using this API.." to "Using this function...". API usually refers to a set of functions/routines(courtesy: Kevin G). Everything else looks good to me.

Comment (deepikabhavnani) [inline docs/api/networksocket/SocketStats.md]:
```suggestion
The `SocketStats` class provides the static function `mbed_stats_socket_get_each` to get the network socket information for all sockets both currently in use and closed. Using this function you can know how many sockets are open, destination IP addresses, the amount of data sent and received in bytes, the type of socket and the timestamp of recent "socket state" changes recorded in `last_change_tick`.
```

Comment (deepikabhavnani) [reply docs/api/networksocket/SocketStats.md]: Done. 👍

File: docs/api/networksocket/SocketStats.md (hunk 2 above)
Comment (kegilbert) [reply docs/api/networksocket/SocketStats.md]: Good point, forgot to check for that earlier. We should add that page as well. Can do it in this PR or another one.

Comment (deepikabhavnani) [reply docs/api/networksocket/SocketStats.md]: Will be good to do it in another one.

File: docs/api/networksocket/SocketStats.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+Use the SocketStats class to read the network socket statistics. SocketStats Class object is created by "InternetSocket" class when statistics are enabled by `nsapi.socket-stats-enable` config option.
+
+The `SocketStats` class provides the static function `mbed_stats_socket_get_each` to get the network socket information for all sockets both currently in use and closed. Using this function you can know how many sockets are open, destination IP addresses, the amount of data sent and received in bytes, the type of socket and the timestamp of recent "socket state" changes recorded in `last_change_tick`.
+
+Socket Statistics are stored in intenal data buffer and maintained for closed sockets as well. In case internal data buffer is full, socket closed with oldest timestamp will be over-written. Maximum amount of sockets statistics stored can be configured by `nsapi.socket-stats-max-count` config option.
+
+### SocketStats class reference
+
+[![View code](https://www.mbed.com/embed/?type=library)](http://os-doc-builder.test.mbed.com/docs/development/mbed-os-api-doxy/)
```
Comment (deepikabhavnani) [inline docs/api/networksocket/SocketStats.md]: @AnotherButler - Network stats PR https://github.com/ARMmbed/mbed-os/pull/8592 is merged, and doxygen for new Class will be generated from features/netsocket/SocketStats.h. Please help me with the link here.

Comment (AnotherButler) [reply docs/api/networksocket/SocketStats.md]: http://os-doc-builder.test.mbed.com/docs/development/mbed-os-api-doxy/class_socket_stats.html

Comment (deepikabhavnani) [inline docs/api/networksocket/SocketStats.md]:
```suggestion
[![View code](https://www.mbed.com/embed/?type=library)](http://os-doc-builder.test.mbed.com/docs/development/mbed-os-api-doxy/class_socket_stats.html)
```

**Comment by deepikabhavnani** [comment]
@AnotherButler - This is ready now, with updated example link.
Please note example is moved from common repo to standalone https://github.com/ARMmbed/mbed-os-example-socket-stats/

**Pull request merged**